File: codegen/results/v002/qwen3.5_4b-nvfp4_think/20260412_085910/typescript/001_csv_to_json/solution.ts
import * as fs from 'fs';
import * as path from 'path';

const csvPath = 'input/data.csv';

// Read the CSV file
const csvContent = fs.readFileSync(csvPath, 'utf8');

// Split into lines and get header
const lines = csvContent.trim().split('\n');
const header = lines[0].split(',');

const data = [];

for (let i = 1; i < lines.length; i++) {
  if (!lines[i].trim()) continue; // skip empty rows
  
  const values = lines[i].split(',');
  
  const row: Record<string, any> = {};
  row.Name = values[0]?.trim() as string;
  row.Age = parseInt(values[1]);
  row.Email = values[2]?.trim() as string;
  row.Score = parseFloat(values[3]);
  
  data.push(row);
}

console.log(JSON.stringify(data));

File: codegen/results/v002/qwen3.5_4b-nvfp4_think/20260412_085910/python/001_csv_to_json/solution.py
import json
import csv

with open('input/data.csv', 'r', encoding='utf-8') as f:
    reader = csv.DictReader(f)
    data = list(reader)

output = []
for row in data:
    output.append({
        "Name": row["Name"],
        "Age": int(row["Age"]),
        "Email": row["Email"],
        "Score": float(row["Score"])
    })

print(json.dumps(output))